scripts/solr_builder/solr_builder/index_subjects.py

```diff
@@ -0,0 +1,101 @@
+import asyncio
+import json
+from asyncio import Future
+from typing import Literal, Set
+
+import httpx
+
+from openlibrary.solr.update_work import build_subject_doc, solr_insert_documents
+from scripts.solr_builder.solr_builder.fn_to_cli import FnToCLI
+from scripts.solr_builder.solr_builder.solr_builder_main import safeget
+
+
+async def index_subjects(
+        subject_type: Literal['subject', 'person', 'place', 'time'],
+        offset=0,
+        limit=1,
+        solr_base_url='http://solr:8983/solr/openlibrary',
+        skip_id_check=False
+):
+    """
+    :return: Returns number of rows added
+    """
+    print(json.dumps({'event': 'starting', 'offset': offset}))
+    async with httpx.AsyncClient() as client:
+        resp = (await client.get(
+            f'{solr_base_url}/select',
+            timeout=30,  # Usually <10, but just in case
+            params={
+                'q': 'type:work',
+                'rows': 0,
+                'facet': 'true',
+                'facet.field': f'{subject_type}_facet',
+                'facet.limit': limit,
+                'facet.offset': offset,
+                'facet.sort': 'index',
+                'facet.mincount': 1,
+                'wt': 'json',
+                'json.nl': 'arrarr',
+            })).json()
+    facets = resp['facet_counts']['facet_fields'][f'{subject_type}_facet']
+    docs = [
+        build_subject_doc(subject_type, subject_name, work_count)
+        for [subject_name, work_count] in facets
+    ]
+
+    # TODO: Move commit_way_later into solr_update
+    commit_way_later_dur = 1000 * 60 * 60 * 24 * 5  # 5 days?
```

Comment:
It's would be nice if the value was a var explicitly annotated as `ms`

Reply:
Fixed